[StructuralMechanics] errors in tests rellated to ResponseFunctions

From the nightly report:

compiler warning:
~~~
/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/custom_response_functions/response_utilities/eigenfrequency_response_function_utility.h:342:47: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
~~~

test errors:
~~~
======================================================================
ERROR: test_execution (structural_response_function_test_factory.TestMassResponseFunction)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/tests/structural_response_function_test_factory.py", line 74, in test_execution
    self.assertAlmostEqual(self.value, 1569.9999999999998)
  File "/usr/lib/python3.5/unittest/case.py", line 861, in assertAlmostEqual
    if round(abs(second-first), places) == 0:
TypeError: unsupported operand type(s) for -: 'float' and 'NoneType'

Stdout:
/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/tests/response_function_tests

======================================================================
ERROR: test_execution (structural_response_function_test_factory.TestStrainEnergyResponseFunction)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/tests/structural_response_function_test_factory.py", line 85, in test_execution
    self.assertAlmostEqual(self.value, 8.484005297318718e-05)
  File "/usr/lib/python3.5/unittest/case.py", line 861, in assertAlmostEqual
    if round(abs(second-first), places) == 0:
TypeError: unsupported operand type(s) for -: 'float' and 'NoneType'

Stdout:
/home/ubuntu/Kratos/applications/StructuralMechanicsApplication/tests/response_function_tests
~~~

--- a/applications/StructuralMechanicsApplication/custom_response_functions/response_utilities/eigenfrequency_response_function_utility.h
+++ b/applications/StructuralMechanicsApplication/custom_response_functions/response_utilities/eigenfrequency_response_function_utility.h
@@ -256,7 +256,7 @@ protected:
         ProcessInfo &CurrentProcessInfo = mrModelPart.GetProcessInfo();
 
         // Predetermine all necessary eigenvalues and prefactors for gradient calculation
-        const unsigned int num_of_traced_eigenfrequencies = mTracedEigenfrequencyIds.size();
+        const std::size_t num_of_traced_eigenfrequencies = mTracedEigenfrequencyIds.size();
         Vector traced_eigenvalues(num_of_traced_eigenfrequencies,0.0);
         Vector gradient_prefactors(num_of_traced_eigenfrequencies,0.0);
         for(std::size_t i = 0; i < num_of_traced_eigenfrequencies; i++)
@@ -275,8 +275,8 @@ protected:
             elem_i.CalculateMassMatrix(mass_matrix_org, CurrentProcessInfo);
             elem_i.CalculateLocalSystem(LHS_org, dummy ,CurrentProcessInfo);
 
-            const unsigned int num_dofs_element = mass_matrix_org.size1();
-            const unsigned int domain_size = CurrentProcessInfo.GetValue(DOMAIN_SIZE);
+            const std::size_t num_dofs_element = mass_matrix_org.size1();
+            const std::size_t domain_size = CurrentProcessInfo.GetValue(DOMAIN_SIZE);
 
             Matrix aux_matrix = Matrix(num_dofs_element,num_dofs_element);
             Vector aux_vector = Vector(num_dofs_element);
@@ -332,14 +332,14 @@ protected:
     {
         rEigenvectorOfElement.resize(size_of_eigenvector,false);
 
-        const unsigned int num_nodes = traced_element.GetGeometry().size();
-        const unsigned int num_node_dofs = size_of_eigenvector/num_nodes;
+        const std::size_t num_nodes = traced_element.GetGeometry().size();
+        const std::size_t num_node_dofs = size_of_eigenvector/num_nodes;
 
         for (std::size_t node_index=0; node_index<num_nodes; node_index++)
         {
             Matrix& rNodeEigenvectors = traced_element.GetGeometry()[node_index].GetValue(EIGENVECTOR_MATRIX);
 
-            for (int dof_index = 0; dof_index < num_node_dofs; dof_index++)
+            for (std::size_t dof_index = 0; dof_index < num_node_dofs; dof_index++)
                 rEigenvectorOfElement(dof_index+num_node_dofs*node_index) = rNodeEigenvectors((eigenfrequency_id-1),dof_index);
         }
     }

--- a/applications/StructuralMechanicsApplication/custom_response_functions/response_utilities/mass_response_function_utility.h
+++ b/applications/StructuralMechanicsApplication/custom_response_functions/response_utilities/mass_response_function_utility.h
@@ -156,7 +156,7 @@ public:
 
 			// Compute total mass of all neighbor elements before finite differencing
 			double mass_before_fd = 0.0;
-			for(unsigned int i = 0; i < ng_elem.size(); i++)
+			for(std::size_t i = 0; i < ng_elem.size(); i++)
 			{
 				Element& ng_elem_i = ng_elem[i];
 				double elem_density = ng_elem_i.GetProperties()[DENSITY];
@@ -172,7 +172,7 @@ public:
 			// Apply pertubation in X-direction and recompute total mass of all neighbor elements
 			double mass_after_fd = 0.0;
 			node_i.X() += mDelta;
-			for(unsigned int i = 0; i < ng_elem.size(); i++)
+			for(std::size_t i = 0; i < ng_elem.size(); i++)
 			{
 				Element& ng_elem_i = ng_elem[i];
 				double elem_density = ng_elem_i.GetProperties()[DENSITY];
@@ -187,7 +187,7 @@ public:
 			// Apply pertubation in Y-direction and recompute total mass of all neighbor elements
 			mass_after_fd = 0.0;
 			node_i.Y() += mDelta;
-			for(unsigned int i = 0; i < ng_elem.size(); i++)
+			for(std::size_t i = 0; i < ng_elem.size(); i++)
 			{
 				Element& ng_elem_i = ng_elem[i];
 				double elem_density = ng_elem_i.GetProperties()[DENSITY];
@@ -202,7 +202,7 @@ public:
 			// Apply pertubation in Z-direction and recompute total mass of all neighbor elements
 			mass_after_fd = 0.0;
 			node_i.Z() += mDelta;
-			for(unsigned int i = 0; i < ng_elem.size(); i++)
+			for(std::size_t i = 0; i < ng_elem.size(); i++)
 			{
 				Element& ng_elem_i = ng_elem[i];
 				double elem_density = ng_elem_i.GetProperties()[DENSITY];
@@ -237,7 +237,7 @@ public:
 
 			// Compute total mass of all neighbor elements before finite differencing
 			double scaling_factor = 0.0;
-			for(unsigned int i = 0; i < ng_elem.size(); i++)
+			for(std::size_t i = 0; i < ng_elem.size(); i++)
 			{
 				Element& ng_elem_i = ng_elem[i];
 				Element::GeometryType& element_geometry = ng_elem_i.GetGeometry();

--- a/applications/StructuralMechanicsApplication/custom_response_functions/response_utilities/strain_energy_response_function_utility.h
+++ b/applications/StructuralMechanicsApplication/custom_response_functions/response_utilities/strain_energy_response_function_utility.h
@@ -435,7 +435,7 @@ protected:
 			WeakPointerVector<Element >& ng_elem = node_i.GetValue(NEIGHBOUR_ELEMENTS);
 
 			double scaling_factor = 0.0;
-			for(unsigned int i = 0; i < ng_elem.size(); i++)
+			for(std::size_t i = 0; i < ng_elem.size(); i++)
 			{
 				Kratos::Element& ng_elem_i = ng_elem[i];
 				Element::GeometryType& element_geometry = ng_elem_i.GetGeometry();

--- a/applications/StructuralMechanicsApplication/tests/structural_response_function_test_factory.py
+++ b/applications/StructuralMechanicsApplication/tests/structural_response_function_test_factory.py
@@ -52,9 +52,12 @@ class StructuralResponseFunctionTestFactory(KratosUnittest.TestCase):
     def _calculate_response_and_gradient(self):
         # Within this location context:
         with controlledExecutionScope(self.path):
-            self.value = self.response_function.CalculateValue()
+            self.response_function.InitializeSolutionStep()
+            self.response_function.CalculateValue()
+            self.value = self.response_function.GetValue()
             self.response_function.CalculateGradient()
             self.gradient = self.response_function.GetShapeGradient()
+            self.response_function.FinalizeSolutionStep()
 
     def tearDown(self):
         # Within this location context:

